### ISSUE TITLE
Getting TypeError when connecting to rethinkdb

### ISSUE THREAD

**Issue by mattiasewers** [neumino/thinky#474]
I'm getting an error when trying to connect to rethinkdb with thinky:

 `TypeError: must start with number, buffer, array or string` from buffer.js:140
with
`var thinky = require('thinky')();`

**Comment by neumino** [comment]
Whole stacktrace?

**Comment by mattiasewers** [comment]
@neumino gettings this

buffer.js:139
    throw new TypeError('must start with number, buffer, array or string');
    ^

TypeError: must start with number, buffer, array or string
    at fromObject (buffer.js:139:11)
    at new Buffer (buffer.js:58:10)
    at Socket.<anonymous> (/Users/ewers/Desktop/coding/ewers-fullstack/Backend/node_modules/rethinkdbdash/lib/connection.js:113:22)
    at emitNone (events.js:72:20)
    at Socket.emit (events.js:166:7)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1048:10)

**Comment by nicolascoutureau** [comment]
Same error for me.

**Comment by IdanAizikNissim** [comment]
Some error
node v5.10.1
rethinkdbdash@2.2.19
thinky@2.2.7

downgarde rethinkdbdash to 2.2.18 fix the problem
"rethinkdbdash": "^2.2.18",
 "thinky": "^2.2.7",

**Comment by JohnnyCrazy** [comment]
Uhm, looks like 1a7ca966de47aeab603634471694ed727515dcb8 broke this

````js
if (options.authKey != null) {
  if (options.password != null || options.user != null) {
    reject(new Err.ReqlDriverError('Cannot use both authKey and user/password').setOperational());
  }
  this.user = 'admin';
  this.password = options.authKey; // <--- Password is set to Authkey? 
} else {
  this.authKey = options.authKey || r._authKey; // <--- options.AuthKey == null, this doesn't make sense?
  this.user = options.user || null;
}
this.password = options.password || null; // <--- Password will be reset to null if you supplied a authkey but no password?
````

I don't know how auth with rethinkdb works in detail, but that `if`-clause seems kinda broken.

**Comment by mattiasewers** [comment]
@IdanAizikNissim yup that worked, until it's solved this seems to me the go to solution, thanks!

**Comment by tlperkins** [comment]
2016-04-14T17:27:57.058Z - error: [src=node] uncaughtException: must start with number, buffer, array or string
{"date":"Thu Apr 14 2016 13:27:57 GMT-0400 (EDT)","process":{"pid":45025,"uid":1515557,"gid":20,"cwd":"/Users/a515557/Projects/dpcs-portal-schema","execPath":"/Users/a515557/n/bin/node","version":"v4.2.2","argv":["/Users/a515557/n/bin/node","/Users/a515557/Projects/dpcs-portal-schema/app.js"],"memoryUsage":{"rss":84901888,"heapTotal":65971712,"heapUsed":41968216}},"os":{"loadavg":[2.69873046875,2.0439453125,1.93505859375],"uptime":101200},"trace":[{"column":11,"file":"buffer.js","function":"fromObject","line":139,"method":null,"native":false},{"column":10,"file":"buffer.js","function":"new  Buffer","line":58,"method":null,"native":false},{"column":22,"file":"/Users/a515557/Projects/dpcs-portal-schema/node_modules/dpcs-server-scaffold/node_modules/thinky/node_modules/rethinkdbdash/lib/connection.js","function":"","line":113,"method":null,"native":false},{"column":20,"file":"events.js","function":"emitNone","line":72,"method":null,"native":false},{"column":7,"file":"events.js","function":"Socket.emit","line":166,"method":"emit","native":false},{"column":10,"file":"net.js","function":"TCPConnectWrap.afterConnect [as oncomplete]","line":1048,"method":"afterConnect [as oncomplete]","native":false}],"stack":["TypeError: must start with number, buffer, array or string","    at fromObject (buffer.js:139:11)","    at new Buffer (buffer.js:58:10)","    at Socket.<anonymous> (/Users/a515557/Projects/dpcs-portal-schema/node_modules/dpcs-server-scaffold/node_modules/thinky/node_modules/rethinkdbdash/lib/connection.js:113:22)","    at emitNone (events.js:72:20)","    at Socket.emit (events.js:166:7)","    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1048:10)"]}
2016-04-14T17:27:57.058Z - error: [src=node] TypeError must start with number, buffer, array or string
{"message":"must start with number, buffer, array or string","stack":"TypeError: must start with number, buffer, array or string\n    at fromObject (buffer.js:139:11)\n    at new Buffer (buffer.js:58:10)\n    at Socket.<anonymous> (/Users/a515557/Projects/dpcs-portal-schema/node_modules/dpcs-server-scaffold/node_modules/thinky/node_modules/rethinkdbdash/lib/connection.js:113:22)\n    at emitNone (events.js:72:20)\n    at Socket.emit (events.js:166:7)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1048:10)"}

**Comment by csaintc** [comment]
Related issue:
https://github.com/neumino/rethinkdbdash/issues/238

**Comment by neumino** [comment]
I just released rethinkdbdash 2.2.20. If you update thinky, things will work again.
